website/docs/mlflow/mlflow_on_synapse_analytics.md

```diff
@@ -0,0 +1,161 @@
+import os
+os.environ["AZURE_TENANT_ID"] = "<tenantId> in the service principle json above"
+os.environ["AZURE_CLIENT_ID"] = "<clientId> in the service principle json above"
+os.environ["AZURE_CLIENT_SECRET"] = "<clientSecret> in the service principle json above"
+```
+
+You can download the workspace configuration file by:
+
+- Navigate to [Azure ML studio](https://ml.azure.com/home)
+- Click the workspace you want to log your experiment with
+- Click on the upper-right corner of the page -> Download config file.
+
+<img src="https://mmlspark.blob.core.windows.net/graphics/Documentation/getamlconfigfile.png" width="400" style="float: center;"/>
+
+the AML config file will contain the following information
+
+```json
+{
+    "subscription_id": "your subscription_id",
+    "resource_group": "your resource_group",
+    "workspace_name": "your workspace_name"
+}
+```
+
+
+```python
+
+from azure.ai.ml import MLClient
+from azure.identity import DefaultAzureCredential
+
+#Enter details of your AzureML workspace
+subscription_id = "subscription_id in the AML config file above"
+resource_group = "resource_group in the AML config file above"
+workspace_name = "workspace_name in the AML config file above"
+
+ml_client = MLClient(credential=DefaultAzureCredential(),
+                        subscription_id=subscription_id, 
+                        resource_group_name=resource_group,
+                        workspace_name=workspace_name)
+
+mlflow_tracking_uri = ml_client.workspaces.get(ml_client.workspace_name).mlflow_tracking_uri
+mlflow.set_tracking_uri(mlflow_tracking_uri)
+```
+
+## Logging SynapseML LightGBMClassifier model with MLflow
+
+
+```python
+from pyspark.sql import SparkSession
+
+# Bootstrap Spark Session
+spark = SparkSession.builder.getOrCreate()
+
+from synapse.ml.core.platform import *
+df = (
+    spark.read.format("csv")
+    .option("header", True)
+    .option("inferSchema", True)
+    .load(
+        "wasbs://publicwasb@mmlspark.blob.core.windows.net/company_bankruptcy_prediction_data.csv"
+    )
+)
+train, test = df.randomSplit([0.85, 0.15], seed=1)
+from pyspark.ml.feature import VectorAssembler
+
+feature_cols = df.columns[1:]
+featurizer = VectorAssembler(inputCols=feature_cols, outputCol="features")
+train_data = featurizer.transform(train)["Bankrupt?", "features"]
+test_data = featurizer.transform(test)["Bankrupt?", "features"]
+
+
+# MLFlow experiment
+artifact_path = "lightGBM"
+experiment_name = "lightGBM_experiment"
+model_name = "lightGBM-model"
+model_version = 1
+model = LightGBMClassifier(
+    objective="binary", featuresCol="features", labelCol="Bankrupt?", isUnbalance=True
+)
+from synapse.ml.lightgbm import LightGBMClassifier
+
+with mlflow.start_run():
+    fit_model = model.fit(train_data)
+    mlflow.spark.log_model(
+        fit_model, artifact_path=artifact_path, registered_model_name=model_name
+    )
```

Comment:
please combine the vector assembler and LightGBM into a single pipeline and log the whole thing